core/src/main/java/io/questdb/griffin/FunctionParser.java

```diff
@@ -336,6 +342,24 @@ public Function parseFunction(
             if (function != null && function.isConstant() && function.extendedOps() == null) {
                 return functionToConstant(function);
             }
+
+            // we don't wrap function in a memoizer if it is a group by or window function
+            // otherwise SqlCodeGen would not recognize the function as a Window or GroupBy function
```

Comment:
We can introduce `isGrouoby()` and `isWindow()` method for `Function` and replace `instanceof GroupByFunction` or `instanceof WindowFunction`